doc/spire_server.md

```diff
@@ -59,6 +59,7 @@ SPIRE configuration files may be represented in either HCL or JSON. Please see t
 | `default_svid_ttl`          | The default SVID TTL                                                       | 1h                            |
 | `trust_domain`              | The trust domain that this server belongs to                               |                               |
 | `upstream_bundle`           | Include upstream CA certificates in the trust bundle                       | false                         |
+| `max_connection_age`        | The maximum allowed duration of a client connection.                       | infinity                      |
```

Comment:
I understand (and appreciate) the desire to not disrupt existing behavior, but I think this change is pretty uncontroversial and will be beneficial across the board. I would strongly prefer for us to start out with a constant value and see how that goes (as opposed to exposing a knob that operators can turn).

Some supporting thoughts/arguments include:
* We want SPIRE to be as close to "just works" as possible. Default of infinity doesn't quite get there IMO
* Making this configurable means we need to educate the users on how to use this knob, why, and in what cases etc... there are a lot of considerations to be had here. There are certainly values that could be "bad". Also see point 1
* There is no clear use case at the moment for "we need value X over here and value Y over there". At most, there is overhead associated w/ TLS handshakes and DNS requests, but the back-of-the-envelope numbers are relatively pedestrian even at low-ish values (e.g. 3 or 5 minutes).
* Since we get jitter for free, we can expect a relatively even distribution over time, resulting in near-immediate response to new servers coming online (as opposed to having to wait absolutely for the chosen duration)
* Seems very likely that we can choose a value here that will work well in the vast majority of cases
* We can always make it configurable in the future without compatibility worries if/when we see clear use cases for doing so

Three or five minutes seems fairly reasonable to me. Thoughts?

Reply:
Another thought here (though perhaps non-blocking) is why enforce this on the server side? It is not a security control, and the desired effect seems to slant more towards the agent than the server (i.e. we want agents to re-balance themselves periodically, having the server enforce something feels like a roundabout means to that end). The agent is also in a position to do this more gracefully if need be.

Reply:
Good points. I think a default of 3 minutes is a good place to start. For a deployment of 1,000 agents, it results in 5 DNS additional queries per second. This is not much at all, and reasonable to allow for more dynamic routing.

Regarding agent vs. server control, #1192 has a little more information, but the tl;dr is that the recommended way for achieving our goal is to have the server control the agent's behavior using the "max connection agent" concept, rather than having that logic live in the agent. They've even started to remove much of the existing "dns refresh" logic from the agents altogether.

Reply:
> that the recommended way for achieving our goal is to have the server control the agent's behavior using the "max connection agent" concept, rather than having that logic live in the agent. They've even started to remove much of the existing "dns refresh" logic from the agents altogether.

Oh... that is interesting... perhaps they view this as a defensive measure? I was surprised to see that this option doesn't exist on the client side. I guess I concede to the gRPC peoples.